src/main/java/org/recap/controller/SearchRecordsController.java

```diff
@@ -70,6 +74,20 @@ public InstitutionDetailsRepository getInstitutionDetailsRepository() {
         return institutionDetailsRepository;
     }
 
+    @GetMapping("/checkPermission")
+    public boolean searchRecords(HttpServletRequest request) {
+        HttpSession session = request.getSession(false);
+        boolean authenticated = getUserAuthUtil().isAuthenticated(request, RecapConstants.SCSB_SHIRO_SEARCH_URL);
+        if (authenticated) {
```

Comment:
The code for checkPermission seems to be duplicate multiple across the controller, can we create a helper method in the base controller, which takes url as the parameters and returns the status?